docs/BRANCHING.md

```diff
@@ -0,0 +1,51 @@
+# Branching Strategy for releases
+
+## General Considerations
+
+- Generally the master branch is the one where current development is happening.
+- One objective would be to arrive to a point where the master branch can be released very often (every day?).
+  This means the master branch is releasable at all times. I believe this strategy is called **Continuous Delivery**.
+- At this point we don't think it's possible to do that until the testing infrastructure has been improved.
+- Nevertheless, the branching strategy takes this objective into account.
+- To further support this objective, the strategy wil try to minimize the number of long-lasting branches.
+
+## Release numbering
+- example 4.3.0
+- The first number (4 in this example) is incremented for major changes in the software. What major change mean has to be decided.
+- The second number (3 in this example) is incremented for minor changes in the software.
+- The third number (0 in this example) is used for maintenance releases.
+  A maintenance is created if some bug has to be corrected before the next minor release.
+![](BranchingDiagram.png)
+## Feature/fix branches
+
+- Feature/fix branch are created for every feature we want to add to the software (or bug we want to fix).
+- We will call them feature branches for the rest of this document.
+- They stem from master, are usually short-lived and are merged back to master.
+
+## Maintenance/release branches
+- In general we will try not to create long-lived release branches.
+- A long-lived release branch will be created only if any merge to master that came after the previous release is deemed not suitable to go in the maintenance release.
+- For example the change could be too extensive, or introduce an undue risk in the maintenance release.
+  This is true in particular if the testing infrastructure is not sufficient.
+
+## Hotfixes
```

Comment:
It might be useful to explicitly state that we should label an issue as a "hotfix" issue, as discussed, so it's clear when a bug warrants this type of branching flow.

Reply:
Added something to that effect.